date_to_day.py

```diff
@@ -0,0 +1,48 @@
+from datetime import datetime
+import logging
+import os
+
+def gather_input(max_attempts=3):
+    for _ in range(max_attempts):
+        date = input("Enter a date (mm/dd/yyyy): ")
+        logger.info(f"User input: {date}")
+        try:
+            parsed = datetime.strptime(date, "%m/%d/%Y")
+            logger.info(f"Parsed date: {parsed}")
+            return parsed
+        except ValueError:
+            logger.error("Error - Date Format:", exc_info=True)
+            print("Invalid date format. Please try again.")
+
+    message = "Max attempts reached. Please try again."
+    logger.critical(f"{message}")
+    raise ValueError(message)
+
+def calculate_day_of_week(date):
+    day_of_week = date.strftime("%A")
+    logger.info(f"Type of day_of_week: {type(day_of_week)}")
```

Comment:
nit: not sure if this is valuable to log, clutters the log stream.

Reply:
fixed